client/X11/xf_monitor.c

```diff
@@ -239,43 +260,114 @@ BOOL xf_detect_monitors(xfContext* xfc)
 		settings->MonitorDefArray[nmonitors].y = vscreen->monitors[i].area.top;
 		settings->MonitorDefArray[nmonitors].width = MIN(vscreen->monitors[i].area.right - vscreen->monitors[i].area.left + 1, xfc->desktopWidth);
 		settings->MonitorDefArray[nmonitors].height = MIN(vscreen->monitors[i].area.bottom - vscreen->monitors[i].area.top + 1, xfc->desktopHeight);
-		settings->MonitorDefArray[nmonitors].is_primary = vscreen->monitors[i].primary;
+		settings->MonitorDefArray[nmonitors].orig_screen = i;
 
-		primaryMonitor |= vscreen->monitors[i].primary;
 		nmonitors++;
 	}
 
-	settings->MonitorCount = nmonitors;
-
-	vWidth = vHeight = 0;
-	settings->DesktopPosX = maxWidth - 1;
-	settings->DesktopPosY = maxHeight - 1;
```

Comment:
Agreed. That variable is no longer used and should be completely removed to simplify the code.